utils/mil_tools/nodes/label_me_bag.py

```diff
@@ -0,0 +1,268 @@
+          print string.format(*args)
+
+  def _verify_yaml(self):
+      """
+      Called by constructor to ensure YAML follows correct format.
+
+      TODO: Improve this process, perhaps using the YAML library more extensiviely
+      """
+      def verify_attr(obj, attr, objname, throw=True):
+          if not throw:
+              return attr in obj
+          if not attr in obj:
+              raise Exception("{} does not have required attribute {} in config".format(objname, attr))
+      verify_attr(self.config, 'bags', 'Config')
+      self._print('Pulling segments from {} bag(s):', len(self.config['bags']))
+      for bag in self.config['bags']:
+          verify_attr(bag, 'file', 'Bag')
+          verify_attr(bag, 'labeled_file', 'Bag')
+          if not 'file' in bag:
+              raise Exception("Bag does not contain 'file' attribute {}".format(bag))
+          verify_attr(bag, 'segments', bag['file'])
+          self._print('\tFound {} segments for {}:', len(bag['segments']), bag['file'])
+          for segment in bag['segments']:
+              verify_attr(segment, 'start', 'segment', throw=False)
+              verify_attr(segment, 'stop', 'segment', throw=False)
+              verify_attr(segment, 'freq', 'segment', throw=False)
+              verify_attr(segment, 'topic', 'segment')
+              verify_attr(segment, 'name', 'segment')
+              self._print("\t\tFound segment '{}' from topic '{}' from {} to {} every {}",
+                          segment['name'], segment['topic'],
+                          segment['start'] if segment.has_key('start') else 'start',
+                          segment['stop'] if segment.has_key('stop') else 'stop',
+                          str(segment['freq']) + ' seconds' if segment.has_key('freq') else 'frame')
+
+  def read_bags(self):
+      """
+      Crawls through all bags in config YAML, placing images from the specified
+      topics and the specified frequency into corosponding folders within
+      Images of the labelme instalation.
+      
+      These folders will be created as follows if not already:
+      LABELME_DIR/Images/SEGEMENT_NAME/TOPIC_NAME
+      where TOPIC_NAME replaces all / with @ for compatibility with filesystems.
+      """
+      for bag in self.config['bags']:
+          self._print("Opening {}", bag['file'])
+          self._read_bag(bag['file'], bag['segments'])
+
+  def extract_labels(self):
+      """
+      Finds labeled XML files in LabelMe directories determined from the
+      config file, placing parsed annotation data as ros messages in a new bag
+      """
+      self._print("Extracting labels for all bags in config")
+      for bag in self.config['bags']:
+          self._print("\tExtracting LabelMe annotations for {}".format(bag['file']))
+          # TODO: give output bag a default name like INFILE_labeled.bag
+          outfile = bag['labeled_file']
+          self._extract_labels_bag(bag, outfile)
+
+  def print_report(self):
```

Comment:
Can you return the percentage so this is usable by another program? I think I could have the fully annotated bags be automatically assembled by a simple cron job on the server.